seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/job/ClientJobExecutionEnvironment.java

```diff
@@ -153,12 +153,13 @@ private void uploadActionPluginJar(List<Action> actions, Set<ConnectorJarIdentif
     }
 
     public ClientJobProxy execute() throws ExecutionException, InterruptedException {
+        LogicalDag logicalDag = getLogicalDag();
         JobImmutableInformation jobImmutableInformation =
                 new JobImmutableInformation(
                         Long.parseLong(jobConfig.getJobContext().getJobId()),
                         jobConfig.getName(),
                         isStartWithSavePoint,
-                        seaTunnelHazelcastClient.getSerializationService().toData(getLogicalDag()),
+                        seaTunnelHazelcastClient.getSerializationService().toData(logicalDag),
```

Comment:
Can you explain what the change does?

Reply:
I see, you need to add comments

Reply:
In the getLogicalDag() method, there is a logic within the org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser#fillJobConfig that parses the variable job.name from the configuration file. This logic sets the value of job.name into jobConfig.name. Consequently, when constructing the JobImmutableInformation object, the job name is derived from the value stored in job.name.